### DISCUSSION TITLE
ChatGPT Support

### DISCUSSION THREAD

**Discussion by worldoptimizer** [pgmichael/wavenet-for-chrome#65] [Ideas]
I am writing to request a new feature for the Chrome WaveNet plugin that would allow ChatGPT’s responses to be read out loud to users. ChatGPT is a powerful AI-based chatbot that can provide helpful responses to a wide range of questions and prompts. However, some users may prefer to consume this content in an auditory format, or may find it difficult to read text on a screen.

By integrating ChatGPT’s text-to-speech functionality with the Chrome WaveNet plugin, users would be able to listen to ChatGPT’s responses in a natural-sounding voice, making it easier to engage with the platform and access its content. This would be a valuable addition for users who have visual impairments, learning disabilities, or who simply prefer to listen rather than read.

To implement this feature, the WaveNet plugin would need to be coordinated with ChatGPT’s text-to-speech functionality, so that whenever a user submits a question or prompt to ChatGPT, the system could automatically generate a spoken response using WaveNet. This could be enabled by adding a toggle switch or checkbox to the ChatGPT user interface, allowing users to turn TTS functionality on or off as desired.

Overall, I believe that integrating ChatGPT’s text-to-speech functionality with the Chrome WaveNet plugin would be a valuable enhancement for the platform, making it more accessible and user-friendly for a wide range of users. Thank you for considering this feature request, and I look forward to seeing it implemented in the future.

Sincerely,

Max

**Comment by pgmichael** [comment]
Good idea @worldoptimizer! 👀 I don't know when or if I'll implement this, but turning popular AI assistants into a voice assistant is definitely going in the backlog.

One thing to consider, though, is that you can't stream text to Google Cloud Text-to-Speech, so it would have to wait for the first sentence or so to finish before being able to say it aloud.

**Comment by worldoptimizer** [comment]
Thanks for considering this. I think for the Plus subscribers it wouldn't be a problem as the answers return really fast. For the free version or the model previews (like GPT4) it would be slow(er).
